stock_available_to_promise_release/models/stock_move.py

```diff
@@ -603,6 +614,81 @@ def _get_chained_moves_iterator(self, chain_field):
             yield moves
             moves = moves.mapped(chain_field)
 
+    def _return_quantity_in_stock(self, qty_to_return_per_move):
+        """Return a quantity from a list of moves.
+
+        The quantity to return is in the product uom"""
+        moves_to_return = self.browse([m_id for m_id in qty_to_return_per_move.keys()])
+        moves_per_type = groupby(moves_to_return, lambda m: m.picking_type_id)
+        for picking_type, moves_list in moves_per_type:
+            moves = self.browse().concat(*moves_list)
+            pickings = moves.picking_id
+            if not pickings:
+                continue
+            return_type = picking_type.return_picking_type_id
+            wiz_values = {
+                "picking_id": fields.first(pickings).id,
+                "location_id": return_type.default_location_dest_id.id,
+            }
+            product_return_moves = []
+            if not return_type:
+                message = _(
+                    "The operation %(picking_names)s is done and cannot be returned",
+                    picking_names=", ".join(pickings.mapped("name")),
+                )
+                raise UserError(message)
+            for move in moves:
+                # Cannot return an unprocessed move
+                if move.state != "done":
+                    continue
+                product = move.product_id
+                uom = product.uom_id
+                qty_to_return = qty_to_return_per_move.get(move.id, 0)
```

Comment:
Since qty_to_return_per_move is a defaultdict, you don't need `get`
```suggestion
                qty_to_return = qty_to_return_per_move[move.id]
```

Reply:
Hmmm, I see what you mean, but I don't fully agree.
Yes, it is a `defaultdict` here, but nothing prevents another module to use this method, for instance.
`self.env["stock.move"].browse([1,2,3])._return_quantity_in_stock({1: 10, 2: 10})`
This is valid, kinda useful, and would raise an exception.

Reply:
ping @jbaudoux

Reply:
Fine for me.